### PR TITLE
Use testutil in rules subpackage

### PR DESCRIPTION
**Pull request by jhernandezb**
See #3242

### CODE REVIEW

File: rules/manager_test.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func TestAlertingRule(t *testing.T) {
 		if actual[0] == "" {
 			actual = []string{}
 		}
-		testutil.Equals(t, expected, actual)
+		testutil.Assert(t, len(expected) == len(actual), "%d. Number of samples in expected and actual output don't match (%d vs. %d)", i, len(expected), len(actual))
```
Comment (fabxc) [inline rules/manager_test.go]: Is switching from comparing the exact result to just checking the length the right thing to do here?

Comment (jhernandezb) [reply rules/manager_test.go]: Yes, I converted it wrong the first time.
https://github.com/prometheus/prometheus/blob/dev-2.0/rules/manager_test.go#L124

**Comment by grobie** [comment]
@jhernandezb Can you rebase please to fix the conflict.

**Comment by jhernandezb** [comment]
Conflicts resolved

**Pull request merged**